packages/flutter_tools/lib/src/_flutter_manifest.dart

```diff
@@ -0,0 +1,460 @@
+  app,
+
+  /// A Flutter project is considered a module when it has a `module:`
+  /// descriptor. A Flutter module project supports integration into an
+  /// existing host app, and has managed platform host code.
+  ///
+  /// Such a project can be created using `flutter create -t module`.
+  module,
+
+  /// A Flutter project is considered a plugin when it has a `plugin:`
+  /// descriptor. A Flutter plugin project wraps custom Android and/or
+  /// iOS code in a Dart interface for consumption by other Flutter app
+  /// projects.
+  ///
+  /// Such a project can be created using `flutter create -t plugin`.
+  plugin,
+}
+
+/// A wrapper around the `flutter:` section in the `pubspec.yaml` file.
+final class FlutterManifest {
+  FlutterManifest({
+    this.appName = '',
+    this.appVersion = '',
+    this.buildName = '',
+    this.buildNumber = '',
+    this.usesMaterialDesign = false,
+    this.projectType = ProjectType.app,
+    this.assets = const <Uri>[],
+    this.fonts = const <Font>[],
+  });
+
+  /// The string value of the top-level `name` property in the `pubspec.yaml` file.
+  final String appName;
+
+  /// The version String from the `pubspec.yaml` file.
+  /// Can be null if it isn't set or has a wrong format.
+  final String appVersion;
+
+  /// The build version name from the `pubspec.yaml` file.
+  /// Can be null if version isn't set or has a wrong format.
+  final String buildName;
+
+  /// The build version number from the `pubspec.yaml` file.
+  /// Can be null if version isn't set or has a wrong format.
+  final String buildNumber;
+
+  /// True if the project uses Material Design.
+  final bool usesMaterialDesign;
+
+  /// The type of Flutter project.
+  final ProjectType projectType;
+
+  /// The list of assets.
+  final List<Uri> assets;
+
+  /// The list of fonts.
+  final List<Font> fonts;
+
+  @override
+  String toString() {
+    return '$runtimeType(appName: $appName, appVersion: $appVersion, buildName: $buildName, '
+        'buildNumber: $buildNumber, projectType: $projectType)';
+  }
+}
+
+/// A wrapper around each of the sections under `fonts:` in `pubspec.yaml`.
+final class Font {
+  Font(this.familyName, this.fontAssets)
+    : assert(familyName != null),
+      assert(fontAssets != null),
+      assert(fontAssets.isNotEmpty);
+
+  final String familyName;
+  final List<FontAsset> fontAssets;
+
+  Map<String, dynamic> get descriptor {
+    return <String, dynamic>{
+      'family': familyName,
+      'fonts': fontAssets.map<Map<String, dynamic>>((FontAsset a) => a.descriptor).toList(),
+    };
+  }
+
+  @override
+  String toString() => '$runtimeType(family: $familyName, assets: $fontAssets)';
+}
+
+/// A wrapper around each of the nested `fonts:` in `pubspec.yaml`.
+final class FontAsset {
+  FontAsset(this.assetUri, {this.weight, this.style})
+    : assert(assetUri != null);
+
+  final Uri assetUri;
+  final int weight;
+  final String style;
+
+  Map<String, dynamic> get descriptor {
```

Comment:
See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#getters-feel-faster-than-methods